### PR TITLE
ACMS-768: Code refactored for sending Site Studio profiling metrics to Telemetry.

### PR DESCRIPTION
**Pull request by vishalkhode1**
ACMS-768: Code refactored for sending Site Studio profiling metrics to Telemetry.

**Motivation**

Fixes Timezone issue & refactored code to implement it using OOPS.

**Testing steps**
- Install Site using UI and on Site Configuration page, change Country and Timezone.

**Merge requirements**
- [ ] _Major change_
- [ ] Manual testing by a reviewer

### CODE REVIEW

File: tests/src/Kernel/AcquiaTelemetryTest.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    $this->telemetry->setTime('start_time', "2021-07-15 12:00:00 +05:30");
+    $this->telemetry->setTime('end_time', "2021-07-15 08:00:00 +01:00");
+    $this->assertSame(1800, $this->telemetry->calculateTime('start_time', 'end_time'));
+    $this->clearValues(['start_time', 'end_time']);
+  }
+
+  /**
+   * Should throw exception when time difference is negative.
+   */
+  public function testNegativeTimeExpectException() {
```
Comment (chandan-singh7929) [inline tests/src/Kernel/AcquiaTelemetryTest.php:53]: @vishalkhode1 - This look pretty clean, just one suggestion, instead of testing exception shouldn't we simply check for positive value?

**Review by chandan-singh7929 — APPROVED** [review]

File: acquia_cms.profile [hunk 2]
```diff
@@ -382,9 +302,8 @@ function acquia_cms_rebuild_site_studio() {
 function acquia_cms_rebuild_site_studio_finished() {
   // The 'success' parameter means no fatal PHP errors were detected. All
   // other error management should be handled using 'results'.
```
Comment (panshulK) [inline acquia_cms.profile:304]: @vishalkhode1 Can we alter the comments here a bit? Are these comments related to Telemtery class itself?

**Review by panshulK — APPROVED** [review]
@vishalkhode1 

Changes looks good to me. Only a minor comment related change is required.

File: tests/src/Kernel/AcquiaTelemetryTest.php (hunk 1 above)
Comment (vishalkhode1) [reply tests/src/Kernel/AcquiaTelemetryTest.php:53]: Yes, we did check for positive value as well as for the negative value.

File: acquia_cms.profile (hunk 2 above)
Comment (vishalkhode1) [reply acquia_cms.profile:304]: No, that's a general comment on how batch API works. That is, it returns the results array. If array contains key: `error` means, batch doesn't executed successfully and it reported some error.

Comment (vishalkhode1) [reply acquia_cms.profile:304]: But yes, ideally it shouldn't be there.
